src/script/descriptor.cpp

```diff
@@ -512,6 +515,19 @@ class AddressDescriptor final : public DescriptorImpl
 public:
     AddressDescriptor(CTxDestination destination) : DescriptorImpl({}, {}, "addr"), m_destination(std::move(destination)) {}
     bool IsSolvable() const final { return false; }
+
+    Optional<OutputType> GetOutputType() const override
+    {
+        switch (m_destination.which()) {
+            case 1 /* PKHash */:
+            case 2 /* ScriptHash */: return OutputType::LEGACY;
```

Comment:
motivation for having ScriptHash be "LEGACY" which I think means p2pkh in most contexts?

Reply:
What else would it be?

I would consider just a P2SH scriptPubKey to be legacy if I had no information about the redeemScript.

Reply:
ahhhh this is `addr()`, ignore this.